crates/bevy_winit/src/system.rs

```diff
@@ -0,0 +1,245 @@
+use bevy_ecs::{
+    entity::Entity,
+    event::EventReader,
+    prelude::{Added, Changed, With},
+    system::{Commands, NonSendMut, Query, Res},
+};
+use bevy_utils::tracing::{error, info};
+use bevy_window::{
+    Cursor, CursorPosition, PrimaryWindow, Window, WindowClosed, WindowComponents, WindowHandle,
+    WindowMode, WindowPosition, WindowResizeConstraints, WindowResolution, WindowState,
+    WindowTitle,
+};
+use raw_window_handle::HasRawWindowHandle;
+use winit::{
+    dpi::{LogicalSize, PhysicalPosition},
+    event_loop::EventLoopWindowTarget,
+};
+
+use crate::{converters, get_best_videomode, get_fitting_videomode, WinitWindows};
+
+/// System responsible for creating new windows whenever a `Window` component is added
+/// to an entity.
+///
+/// This will default any necessary components if they are not already added.
+pub fn create_window_system(
+    mut commands: Commands,
+    event_loop: &EventLoopWindowTarget<()>,
+    created_windows: Query<(Entity, WindowComponents), Added<Window>>,
+    mut winit_windows: NonSendMut<WinitWindows>,
+) {
+    for (window_entity, components) in &created_windows {
+        if let Some(_) = winit_windows.get_window(window_entity) {
+            // Just a safe guard
+            continue;
+        }
+
+        info!("Creating a new window: {:?}", window_entity);
+
+        let winit_window = winit_windows.create_window(&event_loop, window_entity, &components);
+
+        commands
+            .entity(window_entity)
+            .insert(WindowHandle::new(winit_window.raw_window_handle()));
+
+        // TODO: Fix this
+        #[cfg(target_arch = "wasm32")]
+        {
+            let channel = world.resource_mut::<web_resize::CanvasParentResizeEventChannel>();
+            if create_window_event.descriptor.fit_canvas_to_parent {
+                let selector = if let Some(selector) = &create_window_event.descriptor.canvas {
+                    selector
+                } else {
+                    web_resize::WINIT_CANVAS_SELECTOR
+                };
+                channel.listen_to_selector(create_window_event.entity, selector);
+            }
+        }
+    }
+}
+
+pub fn window_destroyed(
```

Comment:
This should be renamed to "destroy_window" or "despawn_window" instead of "window_destroyed" since the system is responsible for doing it rather than responding to it have happened already